lib/mock/mock-agent.js

```diff
@@ -0,0 +1,139 @@
+'use strict'
+
+const { kClients } = require('../core/symbols')
+const Agent = require('../agent')
+const {
+  kAgent,
+  kMockAgentSet,
+  kMockAgentGet,
+  kDispatches,
+  kIsMockActive,
+  kNetConnect,
+  kGetNetConnect,
+  kOptions,
+  kFactory
+} = require('./mock-symbols')
+const MockClient = require('./mock-client')
+const MockPool = require('./mock-pool')
+const { matchValue, buildMockOptions } = require('./mock-utils')
+const { InvalidArgumentError } = require('../core/errors')
+const Dispatcher = require('../dispatcher')
+
+const WeakRef = global.WeakRef || class CompatWeakRef {
```

Comment:
Just a nit, maybe we can move this into utils in the future? Since we have the same code in `agent`. I would add a to-do comment. Not blocking over this, of course.

Reply:
No problem - added! Note, the original code wasn't exactly the same as in Agent, so I've made these the same and adjusted the mocks to mock the `connected` getter